tendrl/gluster_integration/sds_sync/utilization.py

```diff
@@ -40,27 +40,53 @@ def sync_utilization_details(volumes):
     out = ''
     try:
         out, err = cmd.communicate()
+        
         util_det = json.loads(out)
-        for k, v in util_det.iteritems():
-            volume = voldict[k]
-            volume.usable_capacity = int(v['total'])
-            volume.used_capacity = int(v['used'])
-            volume.pcnt_used = str(v['pcnt_used'])
-            volume.total_inode_capacity = int(v['total_inode'])
-            volume.used_inode_capacity = int(v['used_inode'])
-            volume.pcnt_inode_used = str(v['pcnt_inode_used'])
-            volume.save()
-            cluster_used_capacity += volume.used_capacity
-            cluster_usable_capacity += volume.usable_capacity
-    except(KeyError, TypeError, ValueError, AttributeError):
+        err_dict = json.loads(err)
+
+        if err_dict != '':
+            for k, v in err_dict.iteritems():
+                if k in voldict.keys():                    
+                    logger.log(
```

Comment:
This will log error for all the failed volumes one by one. Shouldn't we merge all the failed volumes in one error and log?
Also this happens every sync time and few volumes keep failing always you would hog the log files this way.